### ISSUE TITLE
[Bug]: Load failed with error `failed to load collection, err=failed to spawn replica for collection[nodes not enough]` after upgrade from v2.2.5 to master-20230420-a2ab94ec

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#23589]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:v2.2.5 --> master-20230420-a2ab94ec
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):kafka    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

It would be failed when replicas is 2
all query nodes work well:
 
![image](https://user-images.githubusercontent.com/12268675/233526888-a2a3db3b-f9c1-4860-86fd-936e6397de9a.png)

```
[2023-04-20T10:57:00.795Z] 2023-04-20 10:56:51.525 | INFO     | MainThread |utils:load_and_search:195 - collection name: task_3_IVF_SQ8

[2023-04-20T10:57:00.795Z] 2023-04-20 10:56:51.525 | INFO     | MainThread |utils:load_and_search:196 - load collection

[2023-04-20T10:57:00.795Z] 2023-04-20 10:56:51.525 | INFO     | MainThread |utils:load_and_search:202 - release collection before load if replicas > 1

[2023-04-20T10:57:00.795Z] 2023-04-20 10:56:51.527 | INFO     | MainThread |utils:load_and_search:205 - release time: 0.0016

[2023-04-20T10:57:00.796Z] RPC error: [load_collection], <MilvusException: (code=1, message=failed to load collection, err=failed to spawn replica for collection[nodes not enough])>, <Time:{'RPC start': '2023-04-20 10:56:51.527891', 'RPC error': '2023-04-20 10:56:51.536141'}>

[2023-04-20T10:57:00.796Z] Traceback (most recent call last):

[2023-04-20T10:57:00.796Z]   File "scripts/action_after_upgrade.py", line 111, in <module>

[2023-04-20T10:57:00.796Z]     task_3(data_size, host)

[2023-04-20T10:57:00.796Z]   File "scripts/action_after_upgrade.py", line 58, in task_3

[2023-04-20T10:57:00.796Z]     load_and_search(prefix, replicas=NUM_REPLICAS)

[2023-04-20T10:57:00.796Z]   File "/home/jenkins/agent/workspace/tests/python_client/deploy/scripts/utils.py", line 207, in load_and_search

[2023-04-20T10:57:00.796Z]     c.load(replica_number=replicas)

[2023-04-20T10:57:00.796Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 366, in load

[2023-04-20T10:57:00.796Z]     conn.load_collection(self._name, replica_number=replica_number, timeout=timeout, **kwargs)

[2023-04-20T10:57:00.796Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler

[2023-04-20T10:57:00.796Z]     raise e

[2023-04-20T10:57:00.796Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler

[2023-04-20T10:57:00.796Z]     return func(*args, **kwargs)

[2023-04-20T10:57:00.796Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler

[2023-04-20T10:57:00.796Z]     ret = func(self, *args, **kwargs)

[2023-04-20T10:57:00.796Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 85, in handler

[2023-04-20T10:57:00.796Z]     raise e

[2023-04-20T10:57:00.796Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler

[2023-04-20T10:57:00.796Z]     return func(self, *args, **kwargs)

[2023-04-20T10:57:00.796Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 707, in load_collection

[2023-04-20T10:57:00.796Z]     raise MilvusException(response.error_code, response.reason)

[2023-04-20T10:57:00.796Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=failed to load collection, err=failed to spawn replica for collection[nodes not enough])>
```

### Expected Behavior

all test cases work well

### Milvus Log

![image](https://user-images.githubusercontent.com/12268675/233527267-00957d5a-2c8c-49c2-b196-57b181c82ecd.png)

failed job: deploy_test_kafka_cron/666/
log:
[artifacts-kafka-cluster-upgrade-666-server-second-deployment-logs.tar.gz](https://github.com/milvus-io/milvus/files/11291797/artifacts-kafka-cluster-upgrade-666-server-second-deployment-logs.tar.gz)

[artifacts-kafka-cluster-upgrade-666-server-first-deployment-logs.tar.gz](https://github.com/milvus-io/milvus/files/11291798/artifacts-kafka-cluster-upgrade-666-server-first-deployment-logs.tar.gz)

**Comment by weiliu1031** [comment]
should be fixed by #23415， please verify this

**Comment by yanliang567** [comment]
seems not fixed: #23929

**Comment by zhuwenxing** [comment]
Verification is blocked by https://github.com/milvus-io/milvus/issues/23929
